### PR TITLE
More robust creation of environments

### PR DESCRIPTION
**Pull request by larsrh**
Uses [coursier](https://github.com/alexarchambault/coursier) to locate PIDE jars instead of relying on `sbt-buildinfo`. This works now outside of this SBT project.

### CODE REVIEW

File: setup/src/main/scala/Setup.scala [hunk 1]
```diff
@@ -19,43 +22,78 @@ import acyclic.file
...
+      case Some(platform) =>
+        detectSetup(platform, version) match {
+          case Some(install) =>
+            Future.successful(install)
+          case None =>
+            installTo(platform, version)
+        }
     }
 
+  def fetchImplementation(platform: Platform, version: Version)(implicit ec: ExecutionContext): Future[List[Path]] = {
```
Comment (larsrh) [inline setup/src/main/scala/Setup.scala:62]: @alexarchambault Could you have another look at this function please? Just want to make sure that I'm not doing anything stupid here.

Comment (alexarchambault) [inline setup/src/main/scala/Setup.scala:62]: @larsrh The meta-data (POMs, etc. - the ones used during the resolution) aren't cached here, it seems. It's a matter of setting the `cache` field of the elements in `repositories` below (except `ivy2Local` of course). Be sure that the cache directories of each of them are different.

Comment (larsrh) [inline setup/src/main/scala/Setup.scala:62]: Why do they have to be different?

Comment (larsrh) [inline setup/src/main/scala/Setup.scala:62]: > It's a matter of setting the cache field of the elements in repositories below (except ivy2Local of course)

I did that now in 0220580.

Comment (alexarchambault) [inline setup/src/main/scala/Setup.scala:62]: @larsrh The difference with the way you cache JARs here is that metadata wouldn't be prefixed with the host / base path. They are put directly as _org_`/`_name_/_version_/_filename_ in the cache folders. So if you use one and the same folder everywhere, once all metadata are in cache, they will all appear as coming from the first repository, and JARs will be looked in this repo only (which is likely to fail).

Comment (alexarchambault) [inline setup/src/main/scala/Setup.scala:62]: Also, one last thing to keep in mind: snapshots (if ever you use them) are not deprecated / checked for updates automatically for now. The only way to force updating them for now is to simply remove the corresponding files from the cache, or use the `ForceDownload` `CachePolicy` (which will re-download all it is asked, both snapshots and non-snapshots).

Comment (larsrh) [inline setup/src/main/scala/Setup.scala:62]: Okay, understood. I will separate the file caches and then I think I'm good to merge. Thanks for your assistance!

**Pull request merged**